### ISSUE TITLE
Grammar definition for Array type

### ISSUE THREAD

**Issue by lachlanmcdonald** [Adobe-Marketing-Cloud/htl-spec#60]
The grammar seems to omit an declarative array type, despite one appearing throughout the documentation, i.e. under the [1.2.4. Array Join](https://github.com/Adobe-Marketing-Cloud/htl-spec/blob/master/SPECIFICATION.md#124-array-join) section:

```
${ ["one", "two"] @ join="; " }
```

Perhaps it could be written as:

```
atom = string
     | id
     | int
     | float
     | array
     | bool ;

array = '[' , valueList , ']' ;
```

This would replace the array type that was defined already for `optionValues`, removing `optionValues` and changing:

```
option = id [ , '=' , '(' , exprNode, ')' ]
       | id [ , '=' , atom ] ; 
```
(Note the explicit parentheses around `exprNode` to avoid this from validating:

```
${ @ a=b ? c : d, b=123  }
```

**Comment by raducotescu** [comment]
I think this was corrected through #53. Check the updated `term` element from 939172f.

**Comment by raducotescu** [comment]
@lachlanmcdonald, please reopen the issue if you're not happy with the commit mentioned previously.
